Source/Core/VideoBackends/Metal/MTLVertexFormat.mm

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "VideoBackends/Metal/MTLVertexFormat.h"
+
+#include "VideoCommon/VertexShaderGen.h"
+
+static MTLVertexFormat ConvertFormat(ComponentFormat format, int count, bool int_format)
+{
+  // clang-format off
+  static constexpr MTLVertexFormat formats[2][5][4] = {
```

Comment:
Is there a better way of doing this?

Reply:
Do you mean formatting-wise or logic-wise?  I added an alternative possible formatting in the last commit if you meant formatting.

Logic, we could do an if - switch - switch?
```cpp
if (int_format)
{
  switch (format)
  {
    switch (count)
    {
      case 1: return MTLVertexFormatUCharNormalized;
      case 2: return MTLVertexFormatUChar2Normalized;
      // ...
```

Reply:
Logic-wise. Not sure if I like the 3D array. Maybe the if-switch-switch is better?

Reply:
Something like that?

Reply:
Yeah, that looks better.